psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -2337,6 +2337,106 @@ std::vector<SharedMatrix> MintsHelper::ao_potential_deriv1_helper(int atom) {
     return grad;
 }
 
+std::vector<SharedMatrix> MintsHelper::ao_overlap_half_deriv1_helper(const std::string &half_der_side, int atom) {
+    std::vector<std::string> cartcomp;
```

Comment:
A fixed-size `std::array<std::string, 3>` should suffice here: `std::array<int, 3> cartcomp{ {"X", "Y", "Z"} };`

Reply:
Ditto. Made this change throughout the file.

Reply:
NOTE: I was making this change throughout mintshelper.cc (even in functions that existed before this PR) and found out the hard way that two functions require that the `"x", "y", "z"` in this vector be lowercase, viz. [ao_potential_deriv2_helper](https://github.com/psi4/psi4/blob/ea5b5eb71ab65469629b345d78fe6e7b1a5a34fa/psi4/src/psi4/libmints/mintshelper.cc#L2340-L2344) and [ao_tei_deriv2](https://github.com/psi4/psi4/blob/ea5b5eb71ab65469629b345d78fe6e7b1a5a34fa/psi4/src/psi4/libmints/mintshelper.cc#L2892-L2896). This has to do with the fact that later on in these two functions, they reference a `hess_map` with hard coded key values, e.g. `hess_key["CxAx"]` [here](https://github.com/psi4/psi4/blob/ea5b5eb71ab65469629b345d78fe6e7b1a5a34fa/psi4/src/psi4/libmints/mintshelper.cc#L2451) in which the `"x"` portions of the string are lowercase. If the `"x", "y", "z"` of the `cartomp` vector are uppercase, this becomes an issue [here]() where we build the `key` variable from our `cartcomp` vector and then use it to access different keys of `hess_map`. Using the earlier example, this would build the `key` variable as `"CXAX"`, but `hess_map` only has an entry for the key `"CxAx"`. This is a very minor issue, but it took me a while to hunt down.